spec/controllers/password_resets_controller_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'spec_helper'
+
+describe PasswordResetsController do
+  describe "GET show" do
+    context "with valid token" do
+      
+      let!(:valid_user) { object_generator(:user, token: '123456789') }
+      before { get :show, id: '123456789' }
+      
+      it "renders show template" do
+        expect(response).to render_template :show
+      end
+      
+      it "sets @token" do
+        expect(assigns(:token)).to eq('123456789')
+      end
+    end 
+    
+    context "with invalid token" do
+      it "redirects to expired token page" do
+        get :show, id: '123456789'
+        expect(response).to redirect_to expired_token_path
+      end
+    end 
+  end
+  
+  describe "POST create" do
+    context "with valid token" do
+      
+      context "with password that fails validation" do
+        let!(:valid_user) { object_generator(:user, token: '123456789', password: 'old_password') }
+        before { post :create, token: '123456789', password: 'tiny' }
+        
+        it "renders the password_resets show template" do
+          expect(response).to render_template :show
```

Comment:
If I did a render in that case would I not lose the errors on the password validation?.  I did the redirect on a valid password but thought I had to do a render for error messages on an invalid password, eg if it is too short or blank